File: src/contests/cg/puzzle/vbnet/euclids-algorithm.vb
' 🎮 CodinGame Puzzle - euclids-algorithm
' https://www.codingame.com/training/easy/euclids-algorithm

Imports System

Module Solution
    Sub Main()
        Dim parts() As String = Console.ReadLine().Trim().Split(" "c)
        Dim a As Long = Long.Parse(parts(0))
        Dim b As Long = Long.Parse(parts(1))
        Dim x As Long = a, y As Long = b
        While y <> 0
            Dim q As Long = x \ y
            Dim r As Long = x Mod y
            Console.WriteLine($"{x}={y}*{q}+{r}")
            x = y
            y = r
        End While
        Console.WriteLine($"GCD({a},{b})={x}")
    End Sub
End Module

File: src/contests/cg/puzzle/objectivec/euclids-algorithm.m
// 🎮 CodinGame Puzzle - euclids-algorithm
// https://www.codingame.com/training/easy/euclids-algorithm

#import <Foundation/Foundation.h>

int main() {
    long long a, b;
    scanf("%lld %lld", &a, &b);
    long long x = a, y = b;
    while (y != 0) {
        long long q = x / y, r = x % y;
        printf("%lld=%lld*%lld+%lld\n", x, y, q, r);
        x = y;
        y = r;
    }
    printf("GCD(%lld,%lld)=%lld\n", a, b, x);
    return 0;
}
